src/schedule.js

```diff
@@ -1,14 +1,153 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
 
-import ScheduleEdit, { DAYS } from '../src/schedule-edit'
+import { makeStyles } from '@material-ui/styles'
+import { Button } from 'semantic-ui-react'
+import produce from "immer"
 
 
-const Schedule = () => {
-  const [schedule, update] = useState(DAYS.map(() => Array(24).fill(false)))
+const propTypes = {
+  defaultSchedule: PropTypes.array,
+  onChange: PropTypes.func.isRequired,
+}
+
+const defaultProps = {
+  defaultSchedule: DAYS.map(() => Array(24).fill(false)),
+}
+
+const useStyles = makeStyles({
+  row: {
+    display: 'flex',
+    marginBottom: '1px',
+  },
+  dayButton: {
+    padding: '0.5rem !important',
+    height: '4rem',
+    width: '3rem',
+    flex: '0 0 auto',
+  },
+  hourButton: {
+    marginRight: '1px !important',
+    borderRadius: 'initial !important',
+  }
+})
+
+const DAYS = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
+
+const Schedule = ({ defaultSchedule, onChange }) => {
+  const [schedule, setSchedule] = useState(defaultSchedule)
+  const [dragStart, setDragStart] = useState() // { day, hour }
+  const [hover, setHover] = useState() // { day, hour }
+  
+  useEffect(() => {
+    onChange(schedule)
+  }, [schedule])
+
+  let dragstartState
+  if (dragStart) {
+    dragstartState = schedule[dragStart.day][dragStart.hour]
+  }
+
+  const onDayClick = (_, data) => {
+    const { day } = data.data
+    setSchedule(produce(draftSchedule => {
+      if (draftSchedule[day].filter(hour => hour).length < 24) {
+        draftSchedule[day] = Array(24).fill(true)
```

Comment:
if true then true is 👎 (partially my fault, I tweaked it at some point but fully reverted as I reverted code to original state)

```js
draftSchedule[day] = Array(24).fill(draftSchedule[day].filter(hour => hour).length < 24)
```